internal/xdscache/v3/listener.go

```diff
@@ -598,11 +598,23 @@ func (v *listenerVisitor) visit(vertex dag.Vertex) {
 			var authFilter *http.HttpFilter
 
 			if vh.AuthorizationService != nil {
-				authFilter = envoy_v3.FilterExternalAuthz(
-					vh.AuthorizationService.Name,
-					vh.AuthorizationFailOpen,
-					vh.AuthorizationResponseTimeout,
-				)
+				if vh.AuthorizationBufferSettingsEnabled {
```

Comment:
if we use a struct pointer to wrap the buffer settings, we could remove the `if` and pass the buffer settings in always and just do a nil check in `FilterExternalAuthz` on whether or not to add buffer settings

Reply:
done